modules/core/src/test/scala/sangria/streaming/FS2IntegrationSpec.scala

```diff
@@ -19,70 +21,78 @@ import scala.concurrent.{Await, Future}
  */
 final class FS2IntegrationSpec extends CatsSuite {
 
-  implicit val cs: ContextShift[IO] =
-    IO.contextShift(global)
-
-  val impl: SubscriptionStream[Stream[IO, *]] =
-    streaming.fs2.fs2SubscriptionStream[IO]
+  def impl(d: Dispatcher[IO]): SubscriptionStream[Stream[IO, *]] =
+    streaming.fs2.fs2SubscriptionStream[IO](d, Async[IO])
 
   test("support itself") {
-    impl.supported(streaming.fs2.fs2SubscriptionStream[IO]) should be (true)
+    Dispatcher[IO].use { d =>
+      IO.pure(impl(d).supported(streaming.fs2.fs2SubscriptionStream[IO](d, Async[IO])))
+    }.unsafeRunSync() should be (true)
   }
 
+  def execStream[A](testCase: SubscriptionStream[Stream[IO, *]] => Stream[IO, A]): List[A] =
+    Dispatcher[IO].use { d => testCase(impl(d)).compile.toList }.unsafeRunSync()
+
   test("map") {
-    res(impl.map(Stream.emits[IO, Int](List(1, 2, 10)))(_ + 1)) should be (List(2, 3, 11))
+    execStream(_.map(Stream.emits[IO, Int](List(1, 2, 10)))(_ + 1)) should be (List(2, 3, 11))
   }
 
   test("singleFuture") {
-    res(impl.singleFuture(Future.successful("foo"))) should be (List("foo"))
+    execStream(_.singleFuture(Future.successful("foo"))) should be (List("foo"))
   }
 
   test("single") {
-    res(impl.single("foo")) should be (List("foo"))
+    execStream(_.single("foo")) should be (List("foo"))
   }
 
   test("mapFuture") {
-    res(impl.mapFuture(Stream.emits[IO, Int](List(1, 2, 10)))(x => Future.successful(x + 1))) should be (List(2, 3, 11))
+    execStream(_.mapFuture(Stream.emits[IO, Int](List(1, 2, 10)))(x => Future.successful(x + 1))) should be (List(2, 3, 11))
   }
 
+  def execFuture[A](testCase: SubscriptionStream[Stream[IO, *]] => Future[A]): A =
+    Dispatcher[IO].use { d =>
+      IO.fromFuture(
+        IO.pure(testCase(impl(d)))
+      )
+    }.unsafeRunSync()
+
   test("first") {
-    res(impl.first(Stream.emits[IO, Int](List(1, 2, 3)))) should be (1)
+    execFuture(_.first(Stream.emits[IO, Int](List(1, 2, 3)))) should be (1)
   }
 
   test("first throws an error on empty") {
-    an[NoSuchElementException] should be thrownBy res(impl.first(Stream.empty))
+    an[NoSuchElementException] should be thrownBy execFuture(_.first(Stream.empty))
   }
 
   test("failed") {
-    an [IllegalStateException] should be thrownBy res(impl.failed(new IllegalStateException("foo")))
+    an [IllegalStateException] should be thrownBy execStream(_.failed(new IllegalStateException("foo")))
   }
 
-  test("onComplete handles success") {
-    val stream  = Stream.empty
-    val count   = new AtomicInteger(0)
+  def execOnComplete(s: Stream[IO, _root_.fs2.INothing]): Assertion = {
```

Comment:
`INothing`?

Reply:
```
  /** Alias for `Nothing` which works better with type inference.
    */
  type INothing <: Nothing
```

These are the "valuess" produced by an empty `Stream`.  The test case here is for the `onComplete` method of the Sangria streaming interfacing so empty stream seemed fine.